File: unit_test/batch_mask_ring_visualization.cpp
// 仅用于演示 Mask 与参考双圆环拟合结果。
//
// 复用 batch_image_video.cpp 的推理、拟合、时序、TXT 和视频保存实现，避免两个
// 测试入口以后出现算法参数不一致。这个编译开关只改变可视化内容：
//   - 保留所有实例 Mask；
//   - 画出所有实例的红色检测框；
//   - 只画 cls0 外圈（青色）和 cls1 中圈（紫色）的最终拟合椭圆；
//   - 不画内孔、中心点、可见弧散点、文字和姿态轴。
//
// 命令行参数和输出目录结构与 batch_image_video 完全相同。
#define BATCH_RING_MASK_ONLY 1
#include "batch_image_video.cpp"

File: unit_test/batch_mask_ring_visualization.py
#!/usr/bin/env python3
"""PT 版 Mask + 检测框 + 双圆环可视化。

推理、椭圆拟合、时序处理、命令行参数及输出格式全部复用 final_version.py，
只把最终画面切换成与 batch_mask_ring_visualization.cpp 一致的精简样式：
实例 Mask、红色检测框、青色 cls0 外圈和紫色 cls1 中圈。

示例：
python unit_test/batch_mask_ring_visualization.py \
  --model best.pt --input-path /data/images --mode images \
  --output-path /data/ring_demo --device 0 --no-show
"""

from final_version import CFG, main


if __name__ == "__main__":
    CFG.ring_mask_visualization = True
    CFG.draw_masks = True
    CFG.window_name = "Mask + Detection Boxes + Ring Ellipses"
    raise SystemExit(main())
